sympy/ntheory/tests/test_residue.py

```diff
@@ -128,6 +128,24 @@ def test_residue():
     assert is_nthpow_residue(1, 0, 1) is False
     assert is_nthpow_residue(1, 0, 2) is True
     assert is_nthpow_residue(2, 0, 2) is False
+    assert is_nthpow_residue(3, 5, 30) is True
+    assert is_nthpow_residue(21, 11, 20) is True
+    assert is_nthpow_residue(7, 10, 20) is False
+    assert is_nthpow_residue(5, 10, 20) is True
+    assert is_nthpow_residue(3, 10, 48) is False
+    assert is_nthpow_residue(1, 10, 40) is True
+    assert is_nthpow_residue(3, 10, 24) is False
+    assert is_nthpow_residue(1, 10, 24) is True
+    assert is_nthpow_residue(3, 10, 24) is False
+    assert is_nthpow_residue(2, 10, 48) is False
+    assert is_nthpow_residue(81, 3, 972) is False
+    assert is_nthpow_residue(243, 5, 5103) is True
+    assert is_nthpow_residue(243, 3, 1240029) is False
+    x = set([ pow(i, 7, 1024) for i in range(500)])
+    assert [ is_nthpow_residue(a,7,1024) for a in x]
```

Comment:
I would rather test all values mod m  instead of only those known to be residues, if  m  is not too big. For example,
```
    x = set([pow(i, 56, 1024) for i in range(1024)])
    assert set([a for a in range(1024) if is_nthpow_residue(a, 56, 1024)]) == x
```
I would also be a good idea to let the exponent  n and the module  m  be divisible by a power of 2, since 2 is the critical prime.